test/test_functionalization.py

```diff
@@ -1450,19 +1450,6 @@ def forward(self, arg0_1, arg1_1, arg2_1):
     """)  # noqa: B950
 
 
-    def test_mutation_overlapping_mem(self):
```

Comment:
Oh I don't think we can remove this, since the assert that this is testing is meant to prevent silent correctness (if you try to torch.compile a function that mutates a tensor with internal overlap).

What if we just add the check back into `unfold_copy_inverse` (but still keep your new as_strided path)? That will only raise if someone tries to run the view_inverse code, which (so far) only happens when you're fakeifying tensors or using compile. So seems fine to error in those cases?

Reply:
The issue I'm running into is that the related TORCH_CHECK will never be triggered from functionalization after my changes.

**TORCH_CHECK condition**: `!(reapply_views && size > step)`
**Before my changes**: `unfold_inverse()` can get `reapply_views=True`, and that with `size > step` will trigger the error
**After my changes**: `unfold_inverse()` always gets `reapply_views=False` during functionalization; no way to trigger the error

Should we:
* Change the error condition somehow?
* Change the test logic to try triggering the error from a non-functionalization context? In this case, I think I'll have to re-introduce the test in the next PR since I can't call inverses outside of functionalization yet
* ???

Reply:
Hmm. A few options I can think of:

(1) unconditionally raises that assertion, even if `reapply_views` is False. I'm a bit worried about doing this, since you actually can get correct results mutating a tensor with overlapping memory in eager mode if you functionalize with reapply_views=False. People probably run into this case extremely rarely, but it would be wrong to falsely raise an error.

(2) plumb yet-another-arg, e.g. `called_by_functionalization`, that you use to distinguish when to use *_scatter vs. as_strided (that way we can keep the old behavior of always sending `reapply_views=...` based on the current TLS

2 is probably reasonable?

Reply:
alright sounds good! I did 2